iframe-sandbox/src/ipc.ts

```diff
@@ -107,34 +108,12 @@ export function isGuestError(e: object): e is GuestError {
     "stacktrace" in e && typeof e.stacktrace === "string";
 }
 
-const isObject = (source: unknown): source is Record<string, unknown> =>
-  typeof source === "object" && source != null;
 export const isTaskPerform = (source: unknown): source is TaskPerform =>
   isObject(source) &&
-  typeof source?.intent === "string" &&
-  typeof source?.description === "string" &&
-  isObject(source?.input) &&
-  isJSONSchema(source?.output);
-
-export const isJSONSchema = (source: unknown): source is JSONSchema => {
-  if (!isObject(source)) {
-    return false;
-  }
-
-  switch (source?.type) {
-    case "object":
-    case "array":
-    case "string":
-    case "integer":
-    case "number":
-    case "boolean":
-    case "null":
-      return true;
-    default: {
-      return Array.isArray(source?.anyOf);
-    }
-  }
-};
+  "intent" in source && typeof source.intent === "string" &&
```

Comment:
nit: isn't `"..." in source` redundant here, since `isObject(source)` already makes sure this is an object and `typeof source.<..>` would be `"undefined"` if the key doesn't exist?

Reply:
That's what I thought -- the `Record` type (where `Record<number, unknown>` could be an array) allows this but `object` types do not (and had to add it in a few places) -- maybe there's a better way to do this? but we do sometimes explicitly want an object-like (`{}`) things and not an array or a record

Reply:
We'd get a `Property 'intent' does not exist on type 'object'` otherwise

Reply:
> We'd get a `Property 'intent' does not exist on type 'object'` otherwise

This is a typescript check, and we check that x is not null (`isObject(..)`) and TS `null` is not `object` (unlike JS) -- *usually* we want those checks ("property does not exist"), though using `isRecord` previously allowed any key (TIL as well)